### PR TITLE
Create CountryInput component

### PR DESCRIPTION
**Pull request by Aljullu**
Part of #1574.

For now, I made it use countries directly printed in the settings. But as discussed in #1574 and in preparation for #1575 we would like to get the list of countries/states from the API. But I feel we can work on that in a follow-up PR, while keeping this centered around the component & style.

#### Accessibility

- [x] I've tested using only a keyboard (no mouse)
- [x] I've tested using a screen reader
- [x] All animations respect [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
- [x] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

### Screenshots

imatge

### How to test the changes in this Pull Request:
1. Create a post with a _Checkout_ block and modify [this line](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/master/assets/js/blocks/cart-checkout/checkout/block.js#L112) so input fields appear:
```diff
-{ shippingMethods.length > 0 && (
+{ true && (
```
2. Verify the Country input is a select and matches the styles.

### CODE REVIEW

**Comment by Aljullu** [comment]
Thanks for testing @senadir!

> It doesn’t handle long lists well, try enabling all countries and see, both editor and frontend, some overflow must be applied

Good catch, I should have tested with more countries. I added a `300px` max-height (the same we have for the _Attributes Filter_ dropdown) and it should look good. I wonder if it would make sense to upstream these changes to the `<CustomSelect>` Gutenberg component or they are specific to our implementation.

> both editor and frontend

Actually I think we will need to `<Disable>` all form elements in the editor. But that's out of the scope for this PR.

> the styles are broken on the frontend, storefront. Removing list style and overflow as well

That's weird. Did you stop & start Webpack again? This PR includes one new entry file so Webpack must be run after `git checkout` so it gets it.

**Comment by senadir** [comment]
>  I wonder if it would make sense to upstream these changes to the <CustomSelect> Gutenberg component 
Yes! the element uses never included a long list, so they never had to deal with this, but this is worthwhile fixing anyway.

For the background one, rerunning webpack fixed this.

There is still an entities issue, so I think you need to decode values before inserting them?

image

File: src/Assets.php [hunk 1]
```diff
@@ -126,6 +126,8 @@ public static function get_wc_block_data( $settings ) {
 				'productCount'                => array_sum( (array) $product_counts ),
 				'attributes'                  => array_values( wc_get_attribute_taxonomies() ),
 				'wcBlocksAssetUrl'            => plugins_url( 'assets/', __DIR__ ),
+				'shippingCountries'           => wp_json_encode( WC()->countries->get_shipping_countries() ),
+				'allowedCountries'            => wp_json_encode( WC()->countries->get_allowed_countries() ),
```
Comment (nerrad) [inline src/Assets.php]: does this need to be encoded here? I think it's already handled by https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/master/src/Assets/AssetDataRegistry.php#L219.  No need to encode twice?

Comment (Aljullu) [reply src/Assets.php]: Right! Removed in d6eba824dddff3cc3a1785af064b23de79104abd.

**Comment by Aljullu** [comment]
> There is still an entities issue, so I think you need to decode values before inserting them?

Good catch! Fixed in c6928ae3ea14a51b36675d277d43c9d531e719d8.

**Pull request merged**